### ISSUE TITLE
Users expect transport and errors settings to be configured while these are being ignored

### ISSUE THREAD

**Issue by ramonsmits** [Particular/NServiceBus.AzureFunctions.Worker.ServiceBus#210]
Some users try do to things like:

```c#
.UseNServiceBus(configuration =>
{
    var endpointConfiguration = configuration.AdvancedConfiguration;
    endpointConfiguration.License(NServiceBusLicense);
    endpointConfiguration.EnableInstallers();
    endpointConfiguration.SendFailedMessagesTo(errorQueueName);

    var transport = endpointConfiguration.UseTransport<AzureServiceBusTransport>();
    transport.ConnectionString(serviceBusNameSpaceFQDN);
    transport.CustomTokenCredential(new DefaultAzureCredential());

    var recoverability = endpointConfiguration.Recoverability();
    recoverability.Immediate(        customizations: immediate =>        {            mmediate.NumberOfRetries(3);        });
    recoverability.Delayed(        customizations: delayed =>        {            delayed.NumberOfRetries(5).TimeIncrease(TimeSpan.FromSeconds(30));        });

    endpointConfiguration.LimitMessageProcessingConcurrencyTo(numberOfConsumers);

    // Disable the error queue to let NserviceBus native dead-lettering handle repeated failures:

    configuration.DoNotSendMessagesToErrorQueue();

    //@TODO: Add throttling behavior

})
```

The transport and concurrently settings are ignored but users are unaware of this. Same with `DoNotSendMessagesToErrorQueue` which overrides `SendFailedMessagesTo`. These confliciting settings should results in an endpoint not being able to start of even better. Such API's to not even be available when using Azure Functions.

**Comment by SeanFeldman** [comment]
That's be nice.

Regarding sending messages to the error queue, could be possible with middleware.

**Comment by mikeminutillo** [comment]
In https://github.com/Particular/NServiceBus.AzureFunctions.Worker.ServiceBus/issues/302 we are detecting when someone uses APIs to configure the transport in ways that do not function correctly in a serverless context.

The [documentation](https://github.com/Particular/NServiceBus.AzureFunctions.Worker.ServiceBus/blob/master/src/NServiceBus.AzureFunctions.Worker.ServiceBus/ServiceBusTriggeredEndpointConfiguration.cs#L124-L126) for `DoNotSendMessagesToErrorQueue` states

> Disables moving messages to the error queue **even if an error queue name is configured.**

That seems to imply that you are supposed to be allowed to configure both by design.

**Comment by jpalac** [comment]
@ramonsmits given the above comment, is this issue still valid or can it be closed?

**Comment by ramonsmits** [comment]
No, from my pov the Roslyn analyzer is a workaround for a badly designed API. APIs that don't make sense in the context should not even be available.

Either a different endpointconfiguration should be used or the endpoint configuration should not provide these and expose such APIs differently

**Comment by andreasohlund** [comment]
@ramonsmits The new package has made some improvements by exposing dedicated transport definitions that provide only the settings relevant in a serverless environment. The problem with generic endpoint config things like setting concurrency, etc still remains.

I'm not sure it makes sense to move this issue over since we will need to make core changes to address this? what do U think?

**Comment by ramonsmits** [comment]
@andreasohlund The remaining problem (concurrency, installers, recoverability, etc. being silently ignored) is only visible in the serverless package — that's where users hit it and where they'll come looking for it. That it may ultimately require core changes is an implementation detail. The issue should track the user-facing pain, and the package repo is where that pain is perceived. If core work is needed, that can be split off as a linked core issue once the solution direction is clear.
